webrecorder/web-replay-gen
# Enable hybrid build-runtime archive configuration

- Add option to configure archive data at website runtime with `runtimeOnlyArchives`.
- Refactor UI based on `archivePages` into lit components

Example of runtime configuration: https://github.com/webrecorder/web-replay-gen/tree/config-client-runtime/examples/runtime-only-archives#readme
Documentation of configuration updates: https://github.com/webrecorder/web-replay-gen/tree/config-client-runtime#configuration

Resolves #30.

### Manual testing
1. Run `npm i` to get new dependencies
2. Remove `archives` from `wrg-config.json`
3. Specify a path as `runtimeOnlyArchives` in `wrg-config.json`
4. Run `npm run serve`. Verify archives load as expected
5. Update archive JSON in `_site` and reload page without rebuilding. Verify archives update as expected

### Follow-ups
- [ ] https://github.com/webrecorder/web-replay-gen/issues/32
- [x] Decrease size of build (unnecessary node_modules?)
